### DISCUSSION TITLE
ZigStar Multi Tool cross-platform firmware flasher and backup/restore GUI application for any Zigbee adapter based on TI CC2652, CC1352, and CC2538

### DISCUSSION THREAD

**Discussion by Hedda** [Koenkk/zigbee2mqtt#9710] [General]
FYI, while originally made for [ZigStar LAN Gateway](https://zig-star.com/projects/zigbee-gw-lan/), the ZigStar Multi Tool by xyzroe is open source and also supports all Zigbee dongles/sticks and gateways/bridges/hubs with serial-port access as long as based on CC2652, CC1352, and CC2538 chip from Texas Instruments. 

[https://zig-star.com/radio-docs/zigstar-multi-tool/](https://zig-star.com/radio-docs/zigstar-multi-tool/)

[https://github.com/xyzroe/ZigStarGW-MT](https://github.com/xyzroe/ZigStarGW-MT)

"Multi tool is packed into single executable file. You can get started without installing Python and the required modules, and without using the terminal, - just download and run."

[https://github.com/xyzroe/ZigStarGW-MT/releases](https://github.com/xyzroe/ZigStarGW-MT/releases)

* Read / Write / Erase NVRAM memory of Zigbee module
* Write / Verify / Erase firmware of Zigbee module
* Backdoor Serial Loader turns on automatically
* Manual restart of Zigbee module or ESP32 is available

It is meant for users wanting a very simple way to flash TI Z-Stack Zigbee coordinator and Zigbee router firmware with a graphical interface that works on Windows and Mac OS.

[https://github.com/Koenkk/Z-Stack-firmware/](https://github.com/Koenkk/Z-Stack-firmware/)

Windows screenshot:
![image](https://user-images.githubusercontent.com/6320001/142218127-942e4180-54b2-46b1-a6d1-9b89e49cd6cd.png)

Mac OS X screenshot:
![image](https://user-images.githubusercontent.com/6320001/142218172-6e828a7c-7c81-4114-a350-885c1ea11a33.png)

It is based on cc2538-bsl and zigpy-znp command line tools so will otherwise have same functions as those.

[https://github.com/JelmerT/cc2538-bsl](https://github.com/JelmerT/cc2538-bsl)

[https://github.com/zigpy/zigpy-znp/blob/dev/TOOLS.md](https://github.com/zigpy/zigpy-znp/blob/dev/TOOLS.md)

PS: It does also have the same issues as cc2538-bsl and zigpy-znp command line tools, which include issue auto BSL reset support (automatic restart to bootloader mode) as of yet for adapters using CP2102N USB-to-UART chip like the new SONOFF Zigbee 3.0 USB Dongle Plus by ITead:

[https://github.com/xyzroe/ZigStarGW-MT/issues/2](https://github.com/xyzroe/ZigStarGW-MT/issues/2)

**Comment by thiscantbeserious** [comment]
Who owns that domain? Because it links to a fork/another user and not the original one - this one being the fork:

https://github.com/mercenaruss/ZigStarGW-MT (it links to https://github.com/mercenaruss on www.zig-star.com)

Is that an "officially compiled" binary? Otherwise I would be very carefull with that ... would never use anything else then GitHub releases because these Binaries in a Repo can be changed, Releases can't be altered.

**Reply by Hedda** [reply]
> Who owns that domain? Because it links to a fork/another user and not the original one - this one being the fork:
> 
> https://github.com/mercenaruss/ZigStarGW-MT (it links to https://github.com/mercenaruss on [www.zig-star.com](http://www.zig-star.com))
> 
> Is that an "officially compiled" binary? Otherwise, I would be very careful with that ... would never use anything else than GitHub releases because these Binaries in a Repo can be changed, Releases can't be altered.

@mercenaruss is the founder/creator/maker of ZigStar branded hardware and owns https://zig-star.com as official product domain.

https://github.com/mercenaruss/zigstar_gateways

https://github.com/mercenaruss/zigbee-stick-v4

(as well as developer of the 'old' firmware or ZigStar LAN Gateway)

https://github.com/mercenaruss/ZigStarGW-FW

@xyzroe is collaborating and lead developer of the 'new' and "official" firmware for ZigStar LAN Gateway and ZigStar Multi Tool, etc.

https://zig-star.com/radio-docs/zigstar-multi-tool/

https://zig-star.com/radio-docs/flash-zigstargw/

https://github.com/xyzroe/ZigStarGW-FW

https://github.com/xyzroe/ZigStarGW-MT

You can see both @mercenaruss and @xyzroe replies to issues on all those listed GitHub and they are cross-fork each other repos.

PS: Note that @mercenaruss have now opened discussion boards in most of his repos: 

https://github.com/mercenaruss/zigstar_gateways/discussions

https://github.com/mercenaruss/zigbee-stick-v4/discussions

PPS: mercenaruss (Radu) also replied to a copy of the original post as posted in Home Assistant community forum here:

https://community.home-assistant.io/t/zigstar-multi-tool-cross-platform-firmware-flasher-and-backup-restore-gui-application-for-any-zigbee-adapter-based-on-cc2652-cc1352-and-cc2538/357651

**Comment by mercenaruss** [comment]
@Hedda thank you for explanation.
@thiscantbeserious Yes,i'm the owner of zig-star.com.
and yes that officially compiled binary,i pull them from @xyzroe to have in my own repo.
https://github.com/mercenaruss/ZigStarGW-MT is just a fork, for my own tests. Repo in on @xyzroe Github, he is taking care about software part of the project.
I hope @Hedda and my own explanation is enough to trust the source?
Also @Koenkk added this software to https://www.zigbee2mqtt.io/guide/adapters/#flashing-cc1352-cc2652-cc2538-based-adapters .

**Reply by thiscantbeserious** [reply]
Thanks for the explanation, that way its documented and any further questions and doubt like I had can be avoided. ;)

**Reply by Hedda** [reply]
This is often how small startup companies begin, there will always be a learning curve for its founders and its end-users (or fans)  ;)
